system/lib/wasmfs/memory_file_backend.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+// This file defines the memory file backend of the new file system.
+// Current Status: Work in Progress.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#include "backend.h"
+#include "memory_file.h"
+#include <assert.h>
+#include <emscripten/html5.h>
+#include <emscripten/threading.h>
+#include <utility>
+#include <wasi/api.h>
```

Comment:
Why does this include `wasi/api.h`, `utility`, etc.? Some of these seem unneeded.